common/persistence/dataInterfaces.go

```diff
@@ -402,6 +378,13 @@ type (
 		SetVisibilityTimestamp(timestamp time.Time)
 	}
 
+	// TaskListKey is the struct used to identity TaskLists
+	TaskListKey struct {
```

Comment:
This is being added to prevent overloading of TaskListInfo for requests like CompleteTask and also `worker/tasklist/scavenger` used the same construct internally, so that is consolidated to this as well.